controlpanel/frontend/static/javascripts/modules/role-names.js

```diff
@@ -0,0 +1,53 @@
+moj.Modules.roleNames = {
+  id: "role_name",
+  selectId: "role_name-select",
+  formClass: "appRoles",
+  roleListEndpoint: "/parameters/form/role-list.js",
+  roles: null,
+
+  init() {
+    if (document.querySelectorAll("." + this.formClass).length) {
+      this.getRoles().done(() => {
+        this.loadRolesToSelect();
+        this.bindEvents();
+        console.log(this.roles);
+      });
+    }
+    this.$repoSlugTypeahead = $('#' + this.id)
+  },
+
+  bindEvents() {
+    $('input[name=app_type]').change(() => {
+      this.loadRolesToSelect();
+    })
+  },
+
+  getRoles() {
+    return $.get(
+      this.roleListEndpoint,
+      (data) => {
+        this.roles = data;
+      }
+    )
+  },
+
+  loadRolesToSelect() {
+    $("#" + this.selectId).find('option:not([value=""])').remove();
+    let roles = [];
+    if (self.roles !== null) {
+      const appType = $('input[name=app_type]:checked').val();
+      if (appType) {
+        roles = this.roles[appType];
+      } else {
+        roles = this.roles['airflow'].concat(this.roles['webapp']);
+      }
+    }
+    $('.autocomplete__wrapper').remove();
+    $('#role_name-select').attr('id', 'role_name');
```

Comment:
If you assign `$(this.selectId)` to a variable (eg `selectField`), you can avoid this second DOM search. Also, you hard-coded `#role_name-select` here.